Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -1672,24 +1672,14 @@ void WebsiteDataStore::setTrackingPreventionEnabled(bool enabled)
         return;
 
     RELEASE_LOG(Storage, "%p - WebsiteDataStore::setTrackingPreventionEnabled sessionID=%" PRIu64 ", enabled=%d", this, m_sessionID.toUInt64(), enabled);
-    if (enabled) {
-        m_trackingPreventionEnabled = true;
-        
-        resolveDirectoriesIfNecessary();
```

Comment:
Are we sure that directory resolution is always done before we need to access a directory? It seems correct that we should not do this work in this method, but I want to make sure we resolve directories before we need to access them.

Reply:
It looks like we always do this when initializing a new WebProcess, so it seems like this should always be in the correct state, even without this call.

Reply:
`resolveDirectoriesIfNecessary` is called when we get the parameters of the WebsiteDataStore, and that includes when we create web process and when we add session to network process.